lib/Twig/ContainerRuntimeLoader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Lazily loads Twig runtime implementations from a container.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class Twig_ContainerRuntimeLoader implements Twig_RuntimeLoaderInterface
+{
+    private $container;
+    private $logger;
+
+    public function __construct(ContainerInterface $container, LoggerInterface $logger = null)
+    {
+        $this->container = $container;
+        $this->logger = $logger;
+    }
+
+    public function load($class)
+    {
+        if ($this->container->has($class)) {
+            return $this->container->get($class);
+        }
+
+        if (null !== $this->logger) {
+            $this->logger->warning(sprintf('Class "%s" is not configured as a Twig runtime service.', $class));
```

Comment:
I'm not sure we should keep this logging call. If another loader is able to load the runtime, this is just spamming logs (especially at warning level). and if no loader can load it, ``Twig_Environment::getRuntime`` throws an exception with a meaningful message already